prov/sockets/src/sock_rma.c

```diff
@@ -160,6 +160,26 @@ ssize_t sock_ep_rma_readmsg(struct fid_ep *ep, const struct fi_msg_rma *msg,
 	}
 #endif
 
+	if (tx_ctx->domain->attr.mr_mode & FI_MR_RAW) {
+		void *itr, *key_ptr;
+		struct sock_peer_mr_raw_attr *stored_attr;
+		fastlock_acquire(&tx_ctx->domain->lock);
+		for (i = 0; i < msg->rma_iov_count; i++) {
+			itr = rbtFind(tx_ctx->domain->peer_raw_mr_map.rbtree,
+			              (void*)&msg->rma_iov[i].key);
+			if (!itr) {
+				ret = -FI_ENODATA;
+				break;
+			}
+			rbtKeyValue(tx_ctx->domain->peer_raw_mr_map.rbtree, itr, &key_ptr,
+			            (void **) &stored_attr);
+			assert(stored_attr->key_size == tx_ctx->domain->attr.mr_key_size);
+			sock_tx_ctx_write(tx_ctx, stored_attr->raw_key, stored_attr->key_size);
+		}
+		fastlock_release(&tx_ctx->domain->lock);
+		if (ret) goto err;
+	}
```

Comment:
This looks identical to the code below.  Move into a function.

nit: place goto err on separate line from if (ret)